tests/conftest.py

```diff
@@ -83,110 +83,130 @@ def logged_on_user():
         'short_name': 'FOO',
     }
 
+stream_msg_template = {
+    'id': 537286,
+    'sender_full_name': 'Foo Foo',
+    'timestamp': 1520918722,
+    'client': 'website',
+    'recipient_id': 6076,
+    'sender_email': 'foo@zulip.com',
+    'type': 'stream',
+    'sender_realm_str': '',
+    'flags': ['read'],
+    'sender_id': 5140,
+    'content_type': 'text/x-markdown',
+    'stream_id': 205,
+    'subject': 'Test',
+    'reactions': [],
+    'subject_links': [],
+    'avatar_url': '/user_avatars/2/foo.png?x=x&version=2',
+    'is_me_message': False,
+    'sender_short_name': 'foo',
+    'content': 'Stream content here.',
+    'display_recipient': 'PTEST',
+}
+
+pm_template = {
+    'id': 537287,
+    'sender_full_name': 'Foo Foo',
+    'timestamp': 1520918736,
+    'client': 'website',
+    'recipient_id': 5780,
+    'is_me_message': False,
+    'sender_email': 'foo@zulip.com',
+    'flags': ['read'],
+    'sender_id': 5140,
+    'content_type': 'text/x-markdown',
+    'sender_realm_str': '',
+    'subject': '',
+    'reactions': [],
+    'type': 'private',
+    'avatar_url': '/user_avatars/2/foo.png?x=x&version=2',
+    'subject_links': [],
+    'sender_short_name': 'foo',
+    'content': 'Hey PM content here.',
+    'display_recipient': [{
+        'id': 5179,
+        'is_mirror_dummy': False,
+        'full_name': 'Boo Boo',
+        'short_name': 'boo',
+        'email': 'boo@zulip.com',
+    }, {
+        'short_name': 'foo',
+        'id': 5140,
+        'is_mirror_dummy': False,
+        'full_name': 'Foo Foo',
+        'email': 'foo@zulip.com',
+    }],
+}
+
+group_pm_template = {
+    'id': 537288,
+    'sender_full_name': 'Foo Foo',
+    'timestamp': 1520918737,
+    'client': 'website',
+    'recipient_id': 5780,  # FIXME Unsure
+    'is_me_message': False,
+    'sender_email': 'foo@zulip.com',
+    'flags': ['read'],
+    'sender_id': 5140,
+    'content_type': 'text/x-markdown',
+    'sender_realm_str': '',
+    'subject': '',
+    'reactions': [],
+    'type': 'private',
+    'avatar_url': '/user_avatars/2/foo.png?x=x&version=2',
+    'subject_links': [],
+    'sender_short_name': 'foo',
+    'content': 'Hey PM content here again.',
+    'display_recipient': [{
+        'id': 5179,
+        'is_mirror_dummy': False,
+        'full_name': 'Boo Boo',
+        'short_name': 'boo',
+        'email': 'boo@zulip.com',
+    }, {
+        'short_name': 'foo',
+        'id': 5140,
+        'is_mirror_dummy': False,
+        'full_name': 'Foo Foo',
+        'email': 'foo@zulip.com',
+    }, {
+        'short_name': 'bar',
+        'id': 5180,
+        'is_mirror_dummy': False,
+        'full_name': 'Bar Bar',
+        'email': 'bar@zulip.com',
+    }],
+}
 
-@pytest.fixture
+
+@pytest.fixture(params=[
+    stream_msg_template, pm_template, group_pm_template
+])
+def message(request):
+    """
+    Acts as a parametrize fixture for stream msg, pms and group_pms.
+    """
+    return deepcopy(request.param)
```

Comment:
My suggestion would be to actually leave this first commit as a refactoring and move adding the `message` fixture to the separate (next) commit, where it can be used. So you then have 'refactor' and 'add fixture and use it' commits.

Reply:
Updated